scripts/post_install/stub_generation.py

```diff
@@ -0,0 +1,116 @@
+# STD imports
+import sys
+from pathlib import Path
+from importlib import import_module
+import shutil
+import subprocess
+
+# External imports
+try:
+    from mypy.stubgen import parse_options, generate_stubs
+except ImportError:
+    print("Please install mypy using  \"pip install mypy\" to generate pyi stub files for libraries generated by pybind11.")
+    exit(0)
+
+
+def __AddSubmodules(output_path: Path):
+    if output_path.is_dir():
+        with open(str(output_path / "__init__.pyi"), "a") as file_output:
+            file_output.write("\n\n#   ---- start of includes of sub modules --- \n\n")
+
+            for cpp_sub_module_path in output_path.iterdir():
+                if cpp_sub_module_path.is_dir():
+                    cpp_sub_module_name = str(cpp_sub_module_path.relative_to(cpp_sub_module_path.parent))
+                else:
+                    cpp_sub_module_name = str(cpp_sub_module_path.relative_to(cpp_sub_module_path.parent))[:-4]
+                if cpp_sub_module_name != "__init__":
+                    file_output.write("from . import {:s}\n".format(cpp_sub_module_name))
+
+            file_output.write("\n#   ---- end of includes of sub modules --- \n\n")
+
+        for cpp_sub_module_path in output_path.iterdir():
+            __AddSubmodules(cpp_sub_module_path)
+
+
+def __GenerateStubFilesForModule(
+        kratos_library_path: Path,
+        output_path: Path,
+        cpp_module_name: str) -> None:
+    args = ["-o", str(output_path.absolute()), "-p", cpp_module_name]
+    options = parse_options(args)
+    generate_stubs(options)
+
+    # if sub modules are found, include them in the __init__.pyi
+    submodule_path = kratos_library_path / cpp_module_name
+    __AddSubmodules(submodule_path)
+
+    return cpp_module_name
+
+
+def __MoveGeneratedStub(source: Path, destination: Path) -> None:

```

Comment:
do we need this still?

Reply:
I'd like to revisit this later, so I'd keep it until then. Hopefully I'll find a way to make python and binary symbols to coexist.